### PR TITLE
#54: Iconography

### PR DESCRIPTION
**Pull request by JodiWarren**
- Icon macro & styles
- Bring icons which are inline in text onto the text baseline
- Icon examples

### CODE REVIEW

**Comment by JodiWarren** [comment]
@kevinmpowell A first pass at the icon macro. Please let me know if there are more practical applications to test. I couldn't find any in the Sticker Sheet file, but I may have missed some.

File: library/components/icon/icon.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  @include box-sizing(border-box); // just in case some extra styling gets applied to the icon.
+  display: inline-block;
+  height: 1em;
+  fill: currentColor;
+  line-height: 1em;
+  vertical-align: middle;
+  width: 1em;
+}
+
+h1,
```
Comment (kevinmpowell) [inline library/components/icon/icon.scss]: Per our critique convo let's drop lines 11-23

* [ ] Remove lines 11-23 from `icon.scss`

File: library/components/icon/icon.njk [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{% macro icon(
+    icon=false,
```
Comment (kevinmpowell) [inline library/components/icon/icon.njk]: Pick an icon to use as a default here. My goal with nunjucks macros is that they'll render _something_ when called without any arguments. This allows for rapid prototyping and quick sanity checks when using the components.

You can also drop the if statement on line 7 by setting a default icon.

* [ ] Set a default icon and remove the `if` statement on lines 7 & 24

**Comment by JodiWarren** [comment]
@kevinmpowell @jamesmelzer 
I've made the requested changes.

I've set the default icon to 'check', which is a tick mark and hopefully unlikely to cause offence.

File: library/components/icon/icon.njk [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+) }}
+
+param: icon {string} [default: check] The name of the required icon
+param: iconLabel {string} [default: false] An accessible label for the icon. Used if the icon is not decorative. If not set, will use icon name.
+param: decorative {boolean} [default: true] If set to false, will make it visible to screen readers.
+param: class {string} [default: false] Add any required classes in a space-separated list. They will be appended to the default "spirit-icon" class.
+
+ #}
+{% macro icon(
+    icon=check,
```
Comment (kevinmpowell) [inline library/components/icon/icon.njk]: @JodiWarren I'm surprised that `check` isn't a string: `'check'`, I didn't know nunjucks would automatically convert to a string. For clarity, please wrap this in quotes.

* [ ] Wrap default icon name in quotes

Comment (JodiWarren) [reply library/components/icon/icon.njk]: @kevinmpowell Whoops, good catch. That's fixed now.

**Comment by kevinmpowell** [comment]
@jamesmelzer can you review this so it can be merged?

**Pull request merged**